platforms/jvm/war/src/integTest/groovy/org/gradle/api/tasks/bundling/WarTaskIntegrationTest.groovy

```diff
@@ -300,7 +300,7 @@ task assertUnresolved {
 
 task war(type: War) {
     dependsOn assertUnresolved
-    classpath = configurations.conf
+    classpath = configurations.conf // TODO: this setter call bypasses the adapter, and sets the property directly, resolving the configuration
```

Comment:
❌  This means the property upgrade was potentially a breaking change with regard to eager configuration resolution.

Note, changing the test line to the following makes the test pass:
```
    classpath.from(configurations.conf)
```

Reply:
Can you that the same issue as in https://github.com/gradle/gradle/issues/30273?

Reply:
Yep, it looks very related